packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -0,0 +1,517 @@
+        int pivotIndex = -1;
+        double maxSize = double.negativeInfinity;
+        for (int i = 0; i < children.length; i++) {
+          if (children[i].byteSize > maxSize) {
+            maxSize = children[i].byteSize.toDouble();
+            pivotIndex = i;
+          }
+        }
+        return pivotIndex;
+      default:
+        return -1;
+    }
+  }
+
+  /// Divides a given list of treemap nodes into four parts:
+  /// L1, P, L2, L3.
+
+  /// P (pivot) is the treemap node chosen to be the pivot based on the pivot type.
+  /// L1 includes all treemap nodes before the pivot treemap node.
+  /// L2 and L3 combined include all treemap nodes after the pivot treemap node.
+  /// A combination of elements are put into L2 and L3 so that
+  /// the aspect ratio of the pivot cell (P) is as close to 1 as it can be.
+
+  /// Example layout:
+  /// ```
+  /// ----------------------
+  /// |      |  P   |      |
+  /// |      |      |      |
+  /// |  L1  |------|  L3  |
+  /// |      |  L2  |      |
+  /// |      |      |      |
+  /// ----------------------
+  /// ```
+  List<TreemapCell> buildTreemap({
+    @required List<TreemapNode> children,
+    @required double width,
+    @required double height,
+    @required double x,
+    @required double y,
+  }) {
+    final isHorizontalRectangle = width > height;
+
+    final totalSize = computeByteSizeForNodes(
+      nodes: children,
+      startIndex: 0,
+      endIndex: children.length - 1,
+    );
+
+    if (children.isEmpty) {
+      return [];
+    }
+
+    // Sort list of children, descending in size.
+    children.sort((a, b) => b.byteSize.compareTo(a.byteSize));
+
+    if (children.length <= 2) {
+      final positionedChildren = <TreemapCell>[];
+      double offset = isHorizontalRectangle ? x : y;
+
+      for (final child in children) {
+        final ratio = child.byteSize / totalSize;
+        positionedChildren.add(
+          TreemapCell(
+            key: UniqueKey(),
+            x: isHorizontalRectangle ? offset : x,
+            y: isHorizontalRectangle ? y : offset,
+            width: isHorizontalRectangle ? ratio * width : width,
+            height: isHorizontalRectangle ? height : ratio * height,
+            onTap: () => cellOnTap(rootNode.childrenMap[child.name]),
+            node: child,
+            levelsVisible: widget.levelsVisible - 1,
+          ),
+        );
+
+        offset += isHorizontalRectangle ? ratio * width : ratio * height;
+      }
+
+      return positionedChildren;
+    }
+
+    final pivotIndex = computePivot(children);
+
+    final pivotDataReference = children[pivotIndex];
+    final pSize = pivotDataReference.byteSize;
+
+    final list1 = children.sublist(0, pivotIndex);
+    final list1Size = computeByteSizeForNodes(
+      nodes: list1,
+      startIndex: 0,
+      endIndex: list1.length - 1,
+    );
+
+    List<TreemapNode> list2 = [];
+    int list2Size = 0;
+    List<TreemapNode> list3 = [];
+    int list3Size = 0;
```

Comment:
can we rename these "size" objects to "byteSize" (e.g. list3ByteSize) so it is clear we are talking about bytes and not list length?

Reply:
Done